src/cs/usfca/edu/histfavcheckout/controller/HistFavCheckoutHandler.java

```diff
@@ -76,27 +76,16 @@ public User addUser(PrimaryKey id) {
 	
 	public ResponseEntity<?> getTopFavs(int start, int num) {
 		int threshold = 10;
-		Optional<Product> products[] = productRepository.findAll();
+		int arrayLength = num;
+		if(num > threshold) {arrayLength = threshold;}
+		Optional<Product> products[] = productRepository.findTopNFavoritedMovies(arrayLength);
```

Comment:
Good work, But no need for all the code you have below. You can just store your results in a List<Products> and return that list. Storing first in Optional<Product> then iterating through that to get the actual Product object isn't needed here.

So replace all lines from this line below with these two lines:

`List<Product> products = productRepository.findTopNFavoritedMovies(arrayLength);`
`return ResponseEntity.status(HttpStatus.OK).body(products);	`